.github/workflows/tests.yaml

```diff
@@ -0,0 +1,170 @@
+name: Tests
+
+on: [push, pull_request, workflow_dispatch]
+
+jobs:
+  rubocop:
+    name: 'Rubocop linting'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: MSP-Greg/setup-ruby-pkgs@v1
+        with:
+          ruby-version: 2.6
+          bundler-cache: true
+      - name: bundle install
+        run: bundle install --jobs 4 --retry 3
```

Comment:
This is now a repeated install, the bundler-cache does install and cache gems.

Reply:
@olleolleolle

Thanks, you're correct.  Notes to self - quick glances at the code are not good, and always look at the logs...

@alexeevit can you catch thhe above on the update discussed above?

Reply:
Since this is a little hidden, I usually tack on a line comment "bundle install and cache gems" to make this clearer and more discoverable.

Reply:
@olleolleolle but doesn't `with: - bundler-cache: true` covers bundle caching?
https://github.com/puma/puma/pull/2869/files#diff-95557d53bf91069e59d82b9d8fcfaf52ec84a762858983edd5ef3c9b3e4c8191R14

Reply:
In the last CI run it used the cache:

https://github.com/puma/puma/runs/6292533678?check_suite_focus=true

Reply:
I thought maybe it makes sense to use the repository and `vendor/bundle` as artifacts and re-use them in the next jobs

Reply:
@alexeevit Hello!

What I meant was, I usually note the setting like this, to show that it actually does a `bundle install` on its own, no need to issue a second `bundle install` after it.

```yaml
bundler-cache: true # 'bundle install' and cache the gems
```